.github/workflows/ci.yaml

```diff
@@ -60,7 +60,7 @@ jobs:
           target: wasm32-unknown-unknown
       - uses: Swatinem/rust-cache@v1
       - run: cargo clippy --workspace --all-targets --all-features -- -D warnings -W clippy::pedantic
-      - run: cargo clippy --workspace --target wasm32-unknown-unknown --all-features -- -D warnings -W clippy::pedantic
+      - run: cargo clippy --workspace --target wasm32-unknown-unknown --all-features -- -D warnings -W clippy::pedantic -A dead_code
```

Comment:
I am really confused why the second commit makes clippy suddenly complain about dead code ... (see [this failure](https://github.com/charliermarsh/ruff/actions/runs/3904543263/jobs/6670466198)). Anyway once #1816 is merged we can simply change this command to ignore `ruff_cli` and then clippy apparently no longer complains about this.

Reply:
I think it has to do with the fact that the WASM build uses `commands.rs`, but never `cache.rs` based on the codepaths. I've run into this before.

Reply:
Does it work to just move the `allow` into `cache.rs`?

Reply:
Right as I said #1816 will move `cache` out of `ruff` to `ruff_cli` so this will no longer be an issue. For the time being the `-A dead_code` should make the CI go through.

Reply:
Let me check.

Reply:
Yeah totally. I'd just rather put it in the file if it's equally easy. But not a big deal.

Reply:
Fixed by adding `#![cfg_attr(target_family = "wasm", allow(dead_code))]` to `cache.rs` and `diagnostics.rs` :grin:

Reply:
Much appreciated :)